src/pickup.cpp

```diff
@@ -147,20 +147,23 @@ interact_results interact_with_vehicle( vehicle *veh, const tripoint &pos,
         choice = selectmenu.ret;
     }
 
-    switch( static_cast<options>( choice ) ) {
+    auto veh_tool = [&]( const itype_id & obj ) {
+        item pseudo( obj );
+        itype_id ammo = pseudo.ammo_default();
+        if( veh->fuel_left( ammo ) < pseudo.ammo_required() ) {
+            return false;
+        }
+        pseudo.ammo_set( ammo, veh->drain( ammo, pseudo.ammo_required() ) );
+        g->u.invoke_item( &pseudo );
+        pseudo.ammo_consume( pseudo.ammo_required(), g->u.pos() );
+        return true;
+    };
```

Comment:
This function doesn't work for variable charge items, such as water purifier:
* `ammo_required` can't be relied on to provide the number of charges required, since it only depends on the tool itself
* Fails to return non-consumed charges.